app/models/conversion_host.rb

```diff
@@ -274,16 +283,46 @@ def miq_ssh_util_args_manageiq_providers_openstack_cloudmanager_vm
   # +extra_vars+ option should be a hash of key/value pairs which, if present,
   # will be passed to the '-e' flag.
   #
-  def ansible_playbook(playbook, extra_vars = {})
-    command = "ansible-playbook #{playbook} -i #{ipaddress}"
+  def ansible_playbook(playbook, extra_vars = {}, auth_type = nil)
+    task = MiqTask.all.select { |t| t.context_data.present? && t.context_data[:conversion_host_id] == id }.sort_by(&:created_on).last
 
-    extra_vars[:v2v_host_type] = resource.ext_management_system.emstype
-    extra_vars.each { |k, v| command += " -e '#{k}=#{v}'" }
+    host = hostname || ipaddress
 
-    connect_ssh { |ssu| ssu.shell_exec(command) }
+    command = "ansible-playbook #{playbook} --inventory #{host}, --become --extra-vars=\"ansible_ssh_common_args='-o StrictHostKeyChecking=no'\""
+
+    auth = authentication_type(auth_type) || authentications.first
+    command << " --user #{auth.userid}"
+
+    case auth
+    when AuthUseridPassword
+      extra_vars[:ansible_ssh_pass] = auth.password
+    when AuthPrivateKey
+      ssh_private_key_file = Tempfile.new('ansible_key')
+      ssh_private_key_file.write(auth.auth_key)
+      ssh_private_key_file.close
+      command << " --private-key #{ssh_private_key_file.path}"
+    else
+      raise MiqException::MiqInvalidCredentialsError, _("Unknown auth type: #{auth.authtype}")
+    end
+
+    extra_vars.each { |k, v| command << " --extra-vars '#{k}=#{v}'" }
+
+    result = AwesomeSpawn.run(command)
+    raise unless result.exit_status.zero?
   rescue => e
-    _log.error("Ansible playbook '#{playbook}' failed for '#{resource.name}' with [#{e.class}: #{e}]")
+    errormsg = "Ansible playbook '#{playbook}' failed for '#{resource.name}' with [#{e.class}: #{e}]"
+    _log.error(errormsg)
+    task&.error(errormsg)
     raise e
+  ensure
+    unless result.nil?
+      ansible_output_name = playbook.split('/').last.split('.').first
+      task&.update_context(task.context_data.merge!(ansible_output_name => result.output))
+    end
+    if !ssh_private_key_file.nil? && File.exist?(ssh_private_key_file.path)
```

Comment:
Just a style thing but we usually prefer `ssh_private_key_file.present?` instead of `!.nil?`